### DISCUSSION TITLE
Showing emails on timeline

### DISCUSSION THREAD

**Discussion by brunorbf** [deejay-hub/timeline-lwc#178] [Q&A] [answered]
Hi,
Is it possible to show the sent emails on the timeline?
Just installed the time warp but dispite having email on activity feed, they don’t appear on timeline. 
thanks in advance, Bruno.

**Comment by deejay-hub** [comment]
Hi @brunorbf would you be able to give us some screenshots to better understand what you're seeing. For example,

1. Navigate to a Contact
<img width="1906" alt="image" src="https://user-images.githubusercontent.com/7021883/160301502-671bc98f-66b3-4aa5-9c3f-43c8042e2dfa.png">

2. Click 'Compose' then create a subject and click send
<img width="1908" alt="image" src="https://user-images.githubusercontent.com/7021883/160301524-42ea2202-f32c-48a2-88eb-bf3da9bd3e4d.png">

3. Verify I can see the email on the standard activity timeline
<img width="621" alt="image" src="https://user-images.githubusercontent.com/7021883/160301386-25c3252d-256f-4c56-93f0-2bc1fd9b7de5.png">

4. Navigate to Time Warp. Click Refresh and verify you see the same Email
<img width="1910" alt="image" src="https://user-images.githubusercontent.com/7021883/160301449-8b076ee8-09ac-4ac8-8af8-ada64f2ca2eb.png">

It would also be great if you could confirm the following.
a) Are you using [Person Accounts](https://help.salesforce.com/s/articleView?id=sf.account_person.htm&type=5)?
b) Do you have [Enhanced Email](https://help.salesforce.com/s/articleView?id=000313606&type=1) turned on?
c) Are you using [Einstein Activity Capture](https://help.salesforce.com/s/articleView?id=sf.aac_how_it_works.htm&type=5)?
d) Do you have 'Allow Users to Relate Multiple Contacts to Tasks and Events' feature enabled?
<img width="932" alt="image" src="https://user-images.githubusercontent.com/7021883/160301625-d5cec93c-f84f-458a-aa79-7c3c97834ec0.png">

**Comment by deejay-hub** [comment]
Hi @brunorbf just a quick follow up to see if you still wanted any assistance with the issue you raised?

**Comment by Brisingr75** [comment]
Hi @deejay-hub ,

I'm also interested to see emails in accounts timeline layout. 
At the moment emails are only visible at the contact level, how can I push to account level ?

Thanks for your component, it's really help !
Steve

**Comment by deejay-hub** [comment]
@Brisingr75 By default we only show tasks and events that are directly related to the Account record. This means records where the ‘Related To’ field is populated with the Account Id. However the standard activity timeline rolls up other types of tasks and events too. For example if a Contact related to the account has a task that also shows up on the Account even though the Account record isn’t in the ‘Related To’ field.

For indirect relationships you'll need to use OpenActivities and ActivityHistory relationships.
Steps are included in the config guide https://quip.com/6yvoAcBukqZB#dDRACAgmIoY

**Comment by Brisingr75** [comment, marked as answer]
Hey @deejay-hub ,

Thanks a lot ! I have added OpenActivities and ActivityHistory and it works well !

Thanks for the support
Steve

**Comment by schaitraQuicken** [comment]
@deejay-hub  
Loved this component! Is there a way to link up messaging sessions and EAC activity to the timeline?

**Comment by deejay-hub** [comment]
Hi @schaitraQuicken . I am no EAC expert but I believe that sadly Inbox or Einstein Activity Capture data is stored on AWS servers and databases, which are hosted by Amazon Web Services behind an AWS Virtual Private Cloud (VPC) (https://help.salesforce.com/s/articleView?id=000323391&type=1). They are not actually stored on the Salesforce platform. That means they are not available to packages like mine to query (or anyone else).  So that's a long way of saying *No* you cannot see EAC Emails on the Time Warp component.

Messaging sessions are different. They are stored on platform and should work. We have added metadata for Contact and Person Account. Contact is active by default but if you need Person Account support you'll have to activate it.

**Reply by mill1012** [reply]
Do you think it would it be possible create a flow to create tasks for every EAC email?  then display those tasks

**Reply by deejay-hub** [reply]
Unlikely @mill1012 EAC emails are like a quantum impossibility. They exist but the moment you try and use them in flow, reports, apex - they are not there.

EAC emails are a virtualised representation of a query that returns data from an external database to the lightning platform. As such any method normally available to standard and custom objects don't apply.

Salesforce would have to make EAC interactions available first. I have no insights into whether this will ever happen.
